### PR TITLE
Add kubeflow namespace resource to enable istio injection

### PR DESCRIPTION
**Pull request by Bobgy**
**Which issue is resolved by this Pull Request:**
Resolves https://github.com/kubeflow/kubeflow/issues/3866

**Description of your changes:**
Add kubeflow namespace resource with istio injection label to enable automatic sidecar injection.

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate-changed-only`
    3. `make test`

### CODE REVIEW

**Comment by Bobgy** [comment]
@jlewi Can you please review this?

**Comment by Bobgy** [comment]
Test failures don't seem related.
/retest

File: kubeflow-namespace/base/namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
```
Comment (jlewi) [inline kubeflow-namespace/base/namespace.yaml:1]: I think you should include cert-manager as well like
https://github.com/kubeflow/gcp-blueprints/blob/master/kubeflow/instance/kustomize/namespaces/namespaces.yaml

I might suggest the following hierarchy

namespaces/all/kustomization.yaml

This way if we need to create different combinations of namespaces we can do so.

Comment (jlewi) [reply kubeflow-namespace/base/namespace.yaml:1]: Actually, you might need to keep "base" for legacy reasons as kfctl depends on it today.

Comment (Bobgy) [reply kubeflow-namespace/base/namespace.yaml:1]: sure, will change like that

Comment (Bobgy) [reply kubeflow-namespace/base/namespace.yaml]: @jlewi PTAL

**Comment by Bobgy** [comment]
Test failure blocked by https://github.com/kubeflow/manifests/issues/1157

**Comment by jlewi** [comment]
@Bobgy Is your PR rebased on the latest master?
Did you try regenerating the tests by running `make generate`?

**Comment by Bobgy** [comment]
@jlewi Can you lgtm again? tests are passing now

**Pull request merged**